libc-top-half/musl/src/thread/pthread_create.c

```diff
@@ -251,9 +300,13 @@ int __pthread_create(pthread_t *restrict res, const pthread_attr_t *restrict att
 		init_file_lock(__stdin_used);
 		init_file_lock(__stdout_used);
 		init_file_lock(__stderr_used);
+#ifdef __wasilibc_unmodified_upstream
 		__syscall(SYS_rt_sigprocmask, SIG_UNBLOCK, SIGPT_SET, 0, _NSIG/8);
+#endif
 		self->tsd = (void **)__pthread_tsd_main;
+#ifdef __wasilibc_unmodified_upstream
 		__membarrier_init();
+#endif
```

Comment:
__membarrier_init is commented out here, but I see above that the function is still defined, presumably to the "dummy" version since the actual version does a Linux syscall. For tidiness, it'd be good to avoid defining this symbol too.